std/src/main/scala/algebra/std/map.scala

```diff
@@ -130,3 +133,25 @@ class MapRng[K, V](implicit V: Rng[V]) extends MapSemiring[K, V] with Rng[Map[K,
   override def minus(x: Map[K, V], y: Map[K, V]): Map[K, V] =
     subtractMap(x, y)(V.minus)(V.negate)
 }
+
+class MapGenBool[K, V](implicit GBV: GenBool[V], EqV: Eq[V]) extends GenBool[Map[K, V]] {
+  def zero: Map[K, V] = Map.empty
+  
+  def and(a: Map[K, V], b: Map[K, V]): Map[K, V] = (
+    (a.keySet intersect b.keySet)
```

Comment:
for performance you might want:
```scala
(a.keySet intersect b.keySet).iterator
  .map { k => (k, GBV.and(a(k), b(k))) }
  .filter { case (_, v) => EqV.neqv(GBV.zero, v) }
  .toMap
```
putting the iterator early prevents materialization after .map and .filter. Also, note, .map is returning a set here and so if filter, so that is going to be bit slower than the iterator.